ee/hogai/graph/filter_options/nodes.py

```diff
@@ -0,0 +1,294 @@
+        current_filters = str(state.current_filters or {})
+
+        # Handle empty change - provide a helpful default task
+        if not change.strip():
+            change = "Show me all session recordings with default filters"
+
+        # Use injected prompts if available, otherwise fall back to default prompts
+        output_message = chain.invoke(
+            {
+                "core_memory": self.core_memory.text if self.core_memory else "",
+                "groups": self._all_entities,
+                "change": change,
+                "current_filters": current_filters,
+            },
+            config,
+        )
+
+        if not output_message.tool_calls:
+            raise ValueError("No tool calls found in the output message.")
+
+        tool_call = output_message.tool_calls[0]
+        result = AgentAction(tool_call["name"], tool_call["args"], tool_call["id"])
+        intermediate_steps = state.intermediate_steps or []
+
+        # Add the new AI message to the progress log
+        ai_message = LangchainAIMessage(
+            content=output_message.content, tool_calls=output_message.tool_calls, id=output_message.id
+        )
+        return PartialFilterOptionsState(
+            tool_progress_messages=[*progress_messages, ai_message],
+            intermediate_steps=[*intermediate_steps, (result, None)],
+            generated_filter_options=state.generated_filter_options,
+        )
+
+
+class FilterOptionsToolsNode(FilterOptionsBaseNode, ABC):
+    MAX_ITERATIONS = 10  # Maximum number of iterations for the ReAct agent
+
+    def run(self, state: FilterOptionsState, config: RunnableConfig) -> PartialFilterOptionsState:
+        toolkit = FilterOptionsToolkit(self._team)
+        intermediate_steps = state.intermediate_steps or []
+        action, _output = intermediate_steps[-1]
+        input = None
+        output = ""
+        tool_result_msg: list[LangchainToolMessage] = []
+
+        try:
+            input = FilterOptionsTool.model_validate({"name": action.tool, "arguments": action.tool_input})
+        except ValidationError as e:
+            output = str(
+                ChatPromptTemplate.from_template(REACT_PYDANTIC_VALIDATION_EXCEPTION_PROMPT, template_format="mustache")
+                .format_messages(exception=e.errors(include_url=False))[0]
+                .content
+            )
+        else:
+            # First check if we've reached the terminal stage and return the filter options
+            if input.name == "final_answer":
+                # Extract the full response structure
+                full_response = {
+                    "result": input.arguments.result,  # type: ignore
+                    "data": input.arguments.data,  # type: ignore
+                }
+
+                return PartialFilterOptionsState(
+                    generated_filter_options=full_response,
+                    change=state.change,
+                    intermediate_steps=[],
```

Comment:
Nit: you can now reset state with Nones.
```suggestion
                    change=None, # unless you want to keep previous change for some reason
                    intermediate_steps=None,
```